diff --git a/drivers/net/wireless/aic8800-accepted/aic8800_bsp/aic_bsp_8800d.c b/drivers/net/wireless/aic8800-accepted/aic8800_bsp/aic_bsp_8800d.c
--- a/drivers/net/wireless/aic8800-accepted/aic8800_bsp/aic_bsp_8800d.c
+++ b/drivers/net/wireless/aic8800-accepted/aic8800_bsp/aic_bsp_8800d.c
@@ -18,2 +18,2 @@
-#ifdef CONFIG_AIC_INTF_SDIO
+#ifdef AICWF_SDIO_SUPPORT
 #define RAM_FMAC_FW_ADDR            0x00120000
