diff --git a/build/soong/ui/build/soong.go b/build/soong/ui/build/soong.go
--- a/build/soong/ui/build/soong.go
+++ b/build/soong/ui/build/soong.go
@@ -227,6 +227,9 @@
 	commonArgs = append(commonArgs, "-l", filepath.Join(pb.config.FileListDir(), "Android.bp.list"))
 	invocationEnv := make(map[string]string)
+	if goMemLimit := os.Getenv("SOONG_GOMEMLIMIT"); goMemLimit != "" {
+		invocationEnv["GOMEMLIMIT"] = goMemLimit
+	}
 	if pb.debugPort != "" {
 		//debug mode
 		commonArgs = append(commonArgs, "--delve_listen", pb.debugPort,
 			"--delve_path", shared.ResolveDelveBinary())
